### DISCUSSION TITLE
Is there any way to add an external library on TrueNAS Scale?

### DISCUSSION THREAD

**Discussion by LouisLkc** [immich-app/immich#6567] [Q&A]
I tried to install Immich on TrueNAS Scale via the Discover app, everything worked except the external library. I added the correct path and scanned, but it couldn't scan anything in the folder.

**Comment by ixtrader** [comment]
I have the same problem. 
![grafik](https://github.com/immich-app/immich/assets/15313221/024827a5-168d-46a1-a076-96adc930b1f2)
in the immich "Account Settings" I entered a External library with the path /mnt/fotoalbum
![grafik](https://github.com/immich-app/immich/assets/15313221/64b86ce2-bf4b-47ea-9037-66ba47e8bf03)
but no photos where found.

Is there a way to verify if the /mnt/fotoalbum path is containing fotos?

**Comment by kungpaogao** [comment]
not sure if this helps, but here's my config that I got to work

something to note is that it took quite a while for the initial scan to show anything (i thought i configured it wrong and even changed the mount point a couple times)

**TrueNAS Scale Immich settings > `Storage Configuration` > `Additional Storage`**
![Screenshot 2024-02-11 010442](https://github.com/immich-app/immich/assets/11168401/0efdcd26-968b-4c8c-a68e-cdedd2858695)

**Immich > `Administration` > `Users` > `Edit user`**
![Screenshot 2024-02-11 010244](https://github.com/immich-app/immich/assets/11168401/a276b959-9603-4891-b1fb-4628c71e80d8)

**Immich > `Account Settings` > `Libraries`**
![Screenshot 2024-02-11 010406](https://github.com/immich-app/immich/assets/11168401/8a0b77c0-3f9e-4ce5-ba5b-a35eeaad6fdd)

**Reply by Adamherzallh** [reply]
i dont seem to have the external path in edit user nor the libraries in my account settings

**Comment by mtwo** [comment]
I was able to solve this! I had a similar configuration to @ixtrader that was refusing to work until I unchecked the "Read Only" box in the TrueNAS app configuration. After unchecking that the external library scans now succeed.

I had already confirmed that the mount was appearing correctly in Immich, which you can do by clicking the open shell button on the immich-server container in the Workload panel of the app details screen in TrueNAS.

**Reply by RodiAda** [reply]
Hi! Are you using an SMB share or dataset without a share?

**Reply by mtwo** [reply]
Immich is accessing a subfolder of the dataset through a host path reference, and the entire dataset is shared over the network via SMB
